File: Server/src/main/java/server/TripService.java
package server;

import model.domain.Trip;
import repository.ITripRepo;

import java.sql.Time;
import java.util.Collection;

public class TripService {
    private final ITripRepo repo;

    public TripService(ITripRepo repo) {
        this.repo = repo;
    }

    public Collection<Trip> getTouristAttractionTrips(String touristAttraction, Time startTime, Time endTime) {
        return repo.getTouristAttractionTrips(touristAttraction, startTime, endTime);
    }

    public Trip getTrip(int id) {
        return repo.getById(id);
    }
}

File: Server/server/TripService.cs
using System;
using System.Collections.Generic;
using Model;
using Persistence;

namespace Server.server
{
    public class TripService
    {
        private readonly ITripRepo _repo;
        public TripService(ITripRepo repo)
        {
            _repo = repo;
        }
        public IEnumerable<Trip> GetTouristAttractionTrips(string touristAttraction, TimeSpan startTime, TimeSpan endTime) =>
            _repo.GetTouristAttractionTrips(touristAttraction, startTime, endTime);
    }
}